src/mon/OSDMonitor.cc

```diff
@@ -796,14 +796,20 @@ class OSDUtilizationDumper : public CrushTreeDumper::Dumper<F> {
 
   bool get_osd_utilization(int id, int64_t* kb, int64_t* kb_used,
 			   int64_t* kb_avail) const {
+    // we always return true for OSDs as the specific OSDs can be out,
```

Comment:
This is my first thought.

But there is no reason we stop to process the whole bucket if any one of the OSDs under this bucket is abnormal because they are out or broken or something alike. And if they are, reset their stats will be fair enough.